app/models/reports/patient_blood_sugar.rb

```diff
@@ -0,0 +1,16 @@
+module Reports
+  class PatientBloodSugar < Reports::View
+    self.table_name = "reporting_patient_blood_sugars"
+    belongs_to :patient
+
+    enum blood_sugar_risk_state: {
+      bs_below_200: "bs_below_200",
+      bs_200_to_300: "bs_200_to_300",
+      bs_over_300: "bs_over_300"
+    }
```

Comment:
Do we foresee a case where we will need to deal with these states in the context of just blood sugars, without the patient association? This could be broken into two tables, if we feel the need to:
- A `blood_sugar_risk_states` table where these buckets are calculated for all blood sugars
- `reporting_patient_blood_sugars` can join on the latest BS from that table.

Reply:
This enum state is being calculated as a part of the materialized view. Calculating it in the materialized view is the simpler, and also more consistent with the rest of the states we evaluate for reporting.

Another option I considered was to add a risk_state column to blood_sugars (column doesn't get synced), and populating this risk_state on save. This would be similar to the `blood_sugar_risk_states` table above. But I chose not to do this since we don't need the risk states outside the reporting pipeline.

Reply:
Sorry, I meant a `blood_sugar_risk_states` materialised view (not a table). Basically break the bucket definition and the "latest per patient" into two views instead of one.

Reply:
We can pull this view out in the future if it's necessary. Since this is only needed in a reporting view now, adding an extra materialized view will also mean adding an extra when fetching reports and it seems like unnecessary complexity at the moment.